### PR TITLE
Add some documentation to help people easy understand where is the position of podTemplate

### PR DESCRIPTION
**Pull request by tonitech**
Since podTemplate is a field of nodeSets. So the original document makes people confused about the position of podTemplate.

### CODE REVIEW

File: docs/managing-compute-resources.asciidoc [hunk 1]
```diff
@@ -25,7 +25,14 @@ For Elasticsearch objects, make sure to consider the heap size when you set reso
 ----
 spec:
   nodeSets:
-  - podTemplate:
+  - name: elasticsearch
```
Comment (sebgl) [inline docs/managing-compute-resources.asciidoc]: I think we can reuse the [quickstart naming scheme](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-quickstart.html#k8s-deploy-elasticsearch) here and name this nodeSet `default`.

File: docs/managing-compute-resources.asciidoc [hunk 2]
```diff
@@ -25,7 +25,14 @@ For Elasticsearch objects, make sure to consider the heap size when you set reso
...
+    count: 1
+    config:
+      node.master: true
+      node.data: true
+      node.ingest: true
+      node.store.allow_mmap: false
```
Comment (sebgl) [inline docs/managing-compute-resources.asciidoc]: The config seems a bit irrelevant here, would it be readable enough if we remove it? (we still have name, count and podTemplate at that level).

**Comment by charith-elastic** [comment]
Thank you for raising this PR. I am closing this because there has been no activity for a while. The suggestions have been incorporated into the documentation via #2273.
